sentry-android-core/src/test/java/io/sentry/android/core/SentryPerformanceProviderTest.kt

```diff
@@ -0,0 +1,37 @@
+package io.sentry.android.core
+
+import android.content.pm.ProviderInfo
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import kotlin.test.BeforeTest
+import kotlin.test.Test
+import kotlin.test.assertNotNull
+import org.junit.runner.RunWith
+
+@RunWith(AndroidJUnit4::class)
+class SentryPerformanceProviderTest {
+
+    @BeforeTest
+    fun `set up`() {
+        AppStartState.getInstance().resetInstance()
+    }
+
+    @Test
+    fun `provider sets app start`() {
+        val providerInfo = ProviderInfo()
+
+        val mockContext = ContextUtilsTest.createMockContext()
+        providerInfo.authority = AUTHORITY
+
+        val provider = SentryPerformanceProvider()
+        provider.attachInfo(mockContext, providerInfo)
+
+        // done by ActivityLifecycleIntegration so forcing it here
+        AppStartState.getInstance().setAppStartEnd()
+
+        assertNotNull(AppStartState.getInstance().appStartInterval)
```

Comment:
`l`: Maybe something like this would make the test a bit more valuable.
```suggestion
        val providerUptimeMillis = SystemClock.uptimeMillis()
        val provider = SentryPerformanceProvider()
        provider.attachInfo(mockContext, providerInfo)

        val expectedAppStartInterval = 10
        val appStartEndMillis = providerUptimeMillis + expectedAppStartInterval

        // done by ActivityLifecycleIntegration so forcing it here
        AppStartState.getInstance().setAppStartEnd(appStartEndMillis)

        assertTrue(expectedAppStartInterval <= AppStartState.getInstance().appStartInterval ?: 0)
```

Reply:
its flaky, `SentryPerformanceProvider` is already loaded by the Android FW already, hence `SentryPerformanceProvider.appStartMillis` already has a value, so subtracting a value may work but it'd make it probably flaky, its a tricky use case.

Reply:
We could fix that by resetting the value with a method annotated with `@TestOnly`. No strong opinion here. You decide if you think it's worth the effort.

Reply:
good point, done